nix/modules/devShells.nix

```diff
@@ -25,14 +25,21 @@
       devShells = {
         default = self'.devShells.holonix;
         holonix = pkgs.lib.makeOverridable
-          ({ holochainOverrides }: pkgs.mkShell {
+          ({ holochainOverrides }: (if inputs.versions.stub or false then pkgs.mkShell {
+            shellHook = ''
+                echo "This Holochain version is not supported on the current Holonix version. Please migrate to the new Holonix"
+                echo "See the instructions here: https://developer.holochain.org/resources/upgrade/upgrade-new-holonix/"
+                echo "Alternatively, you can use the quickstart guide provided with the updated Holonix: https://github.com/holochain/holonix"
+                exit 1;
+            '';
+          } else lib.warn "\n\nThis version of Holonix is being deprecated, please consider migrating: https://developer.holochain.org/resources/upgrade/upgrade-new-holonix/" pkgs.mkShell {
```

Comment:
For users of the existing shell, print a warning message. I've added some newlines which makes it easier to pick out I think. Please feel free to test it.

Then the shell is entirely usable, this is just a passthrough warning.